app/views/automated_tests/_test_runs.html.erb

```diff
@@ -20,11 +20,18 @@
 
 <%# student results page (with instructor tests released) does not need the button %>
 <% unless @current_role.student? && ran_by == 'instructor' %>
-  <% route = @current_role.instructor? ?
+  <%
+    route = @current_role.instructor? ?
     run_tests_course_result_path(@current_course, @result) :
-    execute_test_run_course_assignment_automated_tests_path(@current_course, @assignment) %>
+    execute_test_run_course_assignment_automated_tests_path(@current_course, @assignment)
+
+    test_run_id = @grouping.test_runs.where(role: @current_role).in_progress.first&.id
```

Comment:
Any time there's a `.where` and `.first` the natural question is what happens when there's more than one (this is important to consider even if it's rare). This is a bit complicated in general, but here you can sort in descending order of `created_at` to get the most recent in-progress test.